src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
-
+        if (checkLoadOfMap()) {
+            table = resizeNodesArray();
+        }
+        if (addNodeToArray(key, value, table)) {
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int hash = (key == null ? 0 : generateHashCodeByKey(key));
+        Node<K, V> node = table[hash];
+        if (node == null) {
+            return null;
+        }
+        if (key == node.key) {
+            return table[hash].value;
+        }
+        while (!node.key.equals(key)) {
+            node = node.next;
+        }
+        return node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int generateHashCodeByKey(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int result = 17;
+        result = 31 * result + Math.abs(key.hashCode());
+        result = result % currentCapacity;
+        return result;
+    }
+
+    private boolean putNodeWithExistedHash(Node<K, V> newNode, Node<K, V> nodeInTable) {
+        Node<K, V> previousNode = null;
+        while (nodeInTable != null) {
+            if (nodeInTable.equals(newNode)) {
+                return false;
+            }
+            previousNode = nodeInTable;
+            nodeInTable = nodeInTable.next;
+        }
+        previousNode.next = newNode;
+        return true;
+    }
+
+    private boolean checkLoadOfMap() {
+        return size == threshold;
+    }
+
+    private Node<K, V>[] resizeNodesArray() {
+        currentCapacity *= 2;
+        threshold = (int) (currentCapacity * DEFAULT_LOAD_FACTOR);
+        return replaceDataToResizedArray();
+    }
+
+    private Node[] replaceDataToResizedArray() {
+        Node[] enlargedArray = new Node[currentCapacity];
+        for (int i = 0; i < table.length; i++) {
+            if (table[i] == null) {
+                continue;
+            }
+            addNodeToArray(table[i].key, table[i].value, enlargedArray);
+            if (table[i].next != null) {
+                table[i] = table[i].next;
+                i--;
+            }
+        }
+        return enlargedArray;
+    }
+
+    private boolean addNodeToArray(K key, V value, Node<K, V>[] nodes) {
+        boolean increasingSize = false;
+        int hash = generateHashCodeByKey(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        if (nodes[hash] == null) {
+            nodes[hash] = newNode;
+            increasingSize = true;
+        } else if (equalsKeys(key, nodes[hash].key)) {
+            nodes[hash].value = value;
+        } else {
+            Node<K, V> nodeInTable = nodes[hash];
+            increasingSize = putNodeWithExistedHash(newNode, nodeInTable);
+        }
+        return increasingSize;
+    }
+
+    private boolean equalsKeys(K key1, K key2) {
+        return key1 == key2 || key1 != null && key1.equals(key2);
+    }
+
+    public static class Node<K, V> {
```

Comment:
we must not use public for inner class. can we put this class upper so we can read code easier?